From 2750a6047185497f17be55abdffc257654bf71e3 Mon Sep 17 00:00:00 2001
From: Ram Dayal <ram.dayal@st.com>
Date: Tue, 26 Nov 2024 10:23:07 +0100
Subject: [PATCH] arm64: dts: st: Add Type-C support to stm32mp215f-dk

Add support for the Type-C driver with ADC monitoring and power
profiles to the stm32mp215f-dk device tree. This includes the
addition of voltage dividers for CC1 and CC2, and a USB Type-C
connector node with appropriate properties.

Change-Id: Ie3e7f5f1cb8f5da41c34990ac53267f3f73d81b3
Signed-off-by: Ram Dayal <ram.dayal@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/419575
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp215f-dk.dts | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp215f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp215f-dk.dts
@@ -191,6 +191,24 @@
 		status = "okay";
 	};
 
+	typec {
+		compatible = "adc-usb-c-connector";
+		io-channels = <&vdiv_cc1>, <&vdiv_cc2>;
+		io-channel-names = "cc1", "cc2";
+
+		connector {
+			compatible = "usb-c-connector";
+			label = "USB-C";
+			data-role = "device";
+			power-role = "sink";
+			port {
+				usb_c_port: endpoint {
+					remote-endpoint = <&usb_c_device_port>;
+				};
+			};
+		};
+	};
+
 	vdiv_cc1: voltage-divider-cc1 {
 		compatible = "voltage-divider";
 		#io-channel-cells = <0>;
@@ -601,4 +619,9 @@
 	dr_mode = "peripheral";
 	usb-role-switch;
 	status = "okay";
+	port {
+		usb_c_device_port: endpoint {
+			remote-endpoint = <&usb_c_port>;
+		};
+	};
 };
